### PR TITLE
Tooltip positioning bug fix. Pyplot hline and vline.

### PR DESCRIPTION
**Pull request by ssunkara1**
Fixes issue #142

### CODE REVIEW

File: bqplot/pyplot.py [hunk 1]
```diff
@@ -346,6 +346,70 @@ def axes(mark=None, options={}, **kwargs):
...
+    """Draws a horizontal line at the given level. By default, draws a line at
+    y=0.
+    Parameters
+    ----------
+    level: float (default: 0)
+        The level at which to draw the horizontal line
+    fig: Figure or None
+        The figure for which the axes should be edited. If the value is None,
+        the current figure is used.
+    preserve_domain: boolean (default: True)
```
Comment (rmenegaux) [inline bqplot/pyplot.py]: @ssunkara1 shouldn't this be `False` by default? You usually want the line to show up even if it lies without the domain no?

File: bqplot/pyplot.py [hunk 2]
```diff
@@ -346,6 +346,70 @@ def axes(mark=None, options={}, **kwargs):
...
+    default_width = kwargs.pop('stroke_width', 1)
+    if fig is None:
+        fig = current_figure()
+    sc_x = fig.scale_x
+    plot([0., 1.], [level, level], scales={'x': sc_x}, preserve_domain={'x': True,
+         'y': preserve_domain}, axes=False, colors=default_colors,
+         stroke_width=default_width, update_context=False)
+
+
+def vline(level=0., fig=None, preserve_domain=True, **kwargs):
```
Comment (dmadeka) [inline bqplot/pyplot.py]: Should the `level` be set to 0 by default? Or should it be midway through the `Figure` scale?

File: examples/Pyplot.ipynb [hunk 3]
```diff
@@ -94,6 +94,37 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "plt.vline()\n",
```
Comment (dmadeka) [inline examples/Pyplot.ipynb]: Can you specify a `level` in one of these? And as an example, pass `styles` to it?

File: bqplot/pyplot.py (hunk 2 above)
Comment (ssunkara1) [inline bqplot/pyplot.py]: I think 0 is a reasonable default. I'm ok with raising an error too, but slightly prefer setting the default value. 
Midway through the `Figure` scale will be hard to do and I feel it is not a very common use to be a default.

File: examples/Pyplot.ipynb (hunk 3 above)
Comment (ssunkara1) [inline examples/Pyplot.ipynb]: @dmadeka Added this. I will make the change you suggested to be able to do `plt.hline(10)` instead of `plt.hline(level=10)`

**Comment by dmadeka** [comment]
@ssunkara1 Pending changes to the default args I'm good to merge

**Comment by dmadeka** [comment]
@ssunkara1 If you can quickly rebase this and fix the defaults, we can aim for a release by Sylvain or Jason's talk

**Comment by ssunkara1** [comment]
@dmadeka removed the defaults and changed the example notebook. Have a look

**Comment by SylvainCorlay** [comment]
Will test and merge this tomorrow.

File: bqplot/pyplot.py [hunk 4]
```diff
@@ -346,6 +346,70 @@ def axes(mark=None, options={}, **kwargs):
...
+    for a in fig.axes:
+        a.grid_lines = value
+
+
+def hline(level, fig=None, preserve_domain=False, **kwargs):
+    """Draws a horizontal line at the given level. By default, draws a line at
+    y=0.
+    Parameters
+    ----------
+    level: float (default: 0)
```
Comment (dmadeka) [inline bqplot/pyplot.py]: Change the doc?

**Comment by dmadeka** [comment]
Already tested. Everything looks good except the doc, which has a default value for level

----- Original Message -----
From: Sylvain Corlay <reply@reply.github.com>
To: bqplot@noreply.github.com
CC: DHRUV MADEKA, mention@noreply.github.com
At: 13-Jul-2016 01:10:34


Will test and merge this tomorrow.

**Comment by ssunkara1** [comment]
@dmadeka changed the doc

File: bqplot/pyplot.py [hunk 5]
```diff
@@ -346,6 +346,70 @@ def axes(mark=None, options={}, **kwargs):
...
+
+    if fig is None:
+        fig = current_figure()
+    for a in fig.axes:
+        a.grid_lines = value
+
+
+def hline(level, fig=None, preserve_domain=False, **kwargs):
+    """Draws a horizontal line at the given level. By default, draws a line at
+    y=0.
```
Comment (dmadeka) [inline bqplot/pyplot.py]: Doc

**Pull request merged**